src/packet.ml

```diff
@@ -364,14 +363,14 @@ module Tls_crypt = struct
     let prefix = encode_protocol proto (Cstruct.lenv [ op_buf; payload ]) in
     let r = Cstruct.concat [ prefix; op_buf; payload ] in
     (* packet, to_encrypt_offset, to_encrypt_length *)
-    (r, Cstruct.length prefix + encrypted_offset, len - encrypted_offset)
+    (r, Cstruct.length prefix + 1 + encrypted_offset, len - encrypted_offset)
```

Comment:
what about "Cstruct.length op_buf" instead of the 1? this would make it easier to understand.

Reply:
I don't quite understand why it is "encrypted_offset" here -- shouldn't it be related to `payload`?

Reply:
There are two occurences of `encrypted_offset`:

1) The `payload` includes the cleartext part of the header, the encrypted part of the header (acks, remote session), the encrypted data (if any) and the cleartext wkc (if `HARD_RESET_CLIENT_V3`). `encrypted_offset` is the offset into `payload` that is (to be) encrypted.

2) In the second occurence we subtract from the length of the payload the length of the cleartext part of the payload (`encrypted_offset`).

The wkc is not considered part of the payload as such even though in the code we do. It's a choice between encoding without wkc, encrypting and then fixing the proto length, or pass the wkc as part of the payload and then returning the range to encrypt. We may need to reconsider this if we want to implement `P_CONTROL_WKC_V1` as in that case we will have a non-empty encrypted data as well as cleartext wkc.

Reply:
thanks, and your comment on the `encrypted_offset` binding makes a lot of sense and helps me to understand this better.